### PR TITLE
Issue 4608 - new Toolbar

### LINKED ISSUE #4608: New bottom toolbar
### Description
In order to accommodate to the new features we need to update the bottom toolbar to  be able to handle new stuff. For this we are redoing it with a more pleasing look and better UX for the users.

### Goals
- [ ] As a user I want to able to know which actions are available regarding the current state of the  viewer

### Tasks
- [ ] Create a new bottom toolbar with the ability to show and hide  extra buttons

### Notes

https://www.figma.com/proto/bRqIl0g9psqD3ISkmxwcNp/3D-Repo?type=design&node-id=2215-53055&t=1kPdVdpeEHrt2P06-0&scaling=min-zoom&page-id=182%3A30991&starting-point-node-id=2210%3A51164&show-proto-sidebar=1

### Related Resources
[MockUp
](https://www.figma.com/file/bRqIl0g9psqD3ISkmxwcNp/3D-Repo?type=design&node-id=2216-53787&mode=design&t=1kPdVdpeEHrt2P06-0)

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #4608

#### Description
There is a new toolbar that uses the new design. It's not fully operative yet, but it's what requested by the base issue.

#### Test cases
Test the toolbar buttons as you would with the old toolbar, everything should still work.

#### Known "bug"
- Some models have hidden meshes that cannot become visible. In this scenarios, clicking on "show all" will not remove such an icon as those items are still not visible. You can check whether that's the case by opening the tree card and toggling once or twice the visibility of the direct children of the model. If some of them never reach the visible state (the eye is dark blue), some items are forever hidden

### CODE REVIEW

**Review by The-Daniel — CHANGES_REQUESTED** [review]

**Review by sanmont3drepo — DISMISSED** [review]

File: frontend/src/v5/ui/routes/viewer/toolbar/buttons/buttonOptionsContainer/clipButtons.component.tsx
Comment (The-Daniel) [inline frontend/src/v5/ui/routes/viewer/toolbar/buttons/buttonOptionsContainer/clipButtons.component.tsx:1]: On staging the clip button can be toggled on and off to show/hide the clipping GUI elements. In this branch you can't do this so the clipping GUI cannot be hidden. Is this intended?

**Review by carmenfan — CHANGES_REQUESTED** [review]
Can't run /azp with conflicts

File: frontend/src/v5/ui/routes/viewer/toolbar/buttons/buttonOptionsContainer/clipButtons.component.tsx
Comment (carmenfan) [reply frontend/src/v5/ui/routes/viewer/toolbar/buttons/buttonOptionsContainer/clipButtons.component.tsx:1]: you should still be able to do this

Comment (carmenfan) [reply frontend/src/v5/ui/routes/viewer/toolbar/buttons/buttonOptionsContainer/clipButtons.component.tsx:1]: Just had a go... indeed, like before, we should be able to click on that button to toggle on/off edit mode of the clipping plane 
image

**Review by carmenfan — CHANGES_REQUESTED** [review]
#### Having colour override imposed by a view is not triggering the tool bar icon to pop up
Going into a view here, you can see that it only triggered the show all button but not the colour override button:
img-04-29-23-06-29

This is also not showing up when we have a colour override imposed by the sequence card

#### Transform reset? (feature request)
It's not in the original scope (sorry), but could we also get a button for transform reset? It's possible for us to move geometry via a sequence and even save it as part of a issues/risks viewpoint, you can find a sample model with such sequence here https://issue-4608.dev.3drepo.io/v5/viewer/DevTeamWorkspace/72d935e5-d15f-4bc4-8433-fb6d79d57301/bc8f82ef-db39-4cb0-a4f9-9bb47aa14dc9
img-04-36-23-06-36

At the moment besides manipulating the sequence card or imposing a different view, there's no way for us to reset the transform, so it would be good if we can add a button for that like we do for colour overrides etc.

**Comment by Amantini1997** [comment]
> #### Having colour override imposed by a view is not triggering the tool bar icon to pop up
> Going into a view here, you can see that it only triggered the show all button but not the colour override button: img-04-29-23-06-29 [ img-04-29-23-06-29 ](https://private-user-images.githubusercontent.com/11945337/287796058-a72f9a0c-75e9-4103-bc5a-c36b1d8f4ca1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzcyMjYsIm5iZiI6MTcwMTc3NjkyNiwicGF0aCI6Ii8xMTk0NTMzNy8yODc3OTYwNTgtYTcyZjlhMGMtNzVlOS00MTAzLWJjNWEtYzM2YjFkOGY0Y2ExLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDExNDg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyYThkODc0MmJhMzE3NDEyYTY3Zjk2YWMxY2YyZjhmZTY0ZjhlZDBjYjM0YmE2NGJiMmQ0MzIzY2E4ZDUzZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h3Um9mWwIsN2gre4ZgGIWStb_hvyB0LWYplb8cTyi3g) [ ](https://private-user-images.githubusercontent.com/11945337/287796058-a72f9a0c-75e9-4103-bc5a-c36b1d8f4ca1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzcyMjYsIm5iZiI6MTcwMTc3NjkyNiwicGF0aCI6Ii8xMTk0NTMzNy8yODc3OTYwNTgtYTcyZjlhMGMtNzVlOS00MTAzLWJjNWEtYzM2YjFkOGY0Y2ExLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDExNDg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyYThkODc0MmJhMzE3NDEyYTY3Zjk2YWMxY2YyZjhmZTY0ZjhlZDBjYjM0YmE2NGJiMmQ0MzIzY2E4ZDUzZDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h3Um9mWwIsN2gre4ZgGIWStb_hvyB0LWYplb8cTyi3g)
> 
> This is also not showing up when we have a colour override imposed by the sequence card
> 
> #### Transform reset? (feature request)
> It's not in the original scope (sorry), but could we also get a button for transform reset? It's possible for us to move geometry via a sequence and even save it as part of a issues/risks viewpoint, you can find a sample model with such sequence here https://issue-4608.dev.3drepo.io/v5/viewer/DevTeamWorkspace/72d935e5-d15f-4bc4-8433-fb6d79d57301/bc8f82ef-db39-4cb0-a4f9-9bb47aa14dc9 img-04-36-23-06-36 [ img-04-36-23-06-36 ](https://private-user-images.githubusercontent.com/11945337/287797393-6946576d-c69c-4abd-a9ff-d3da922546f2.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzcyMjYsIm5iZiI6MTcwMTc3NjkyNiwicGF0aCI6Ii8xMTk0NTMzNy8yODc3OTczOTMtNjk0NjU3NmQtYzY5Yy00YWJkLWE5ZmYtZDNkYTkyMjU0NmYyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDExNDg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5Y2VmMGVhYWFkNGRlOWE0NzdiNDFlNzYyMjFmZWU3ZTBhYzQxNjIyMmMwODI0NGY4ZjU0YTc2Yjc0ZjNkYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bv_in7fiimSuDIjKPBeHkgo3fLmX9YYs55LGfjuOipU) [ ](https://private-user-images.githubusercontent.com/11945337/287797393-6946576d-c69c-4abd-a9ff-d3da922546f2.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3NzcyMjYsIm5iZiI6MTcwMTc3NjkyNiwicGF0aCI6Ii8xMTk0NTMzNy8yODc3OTczOTMtNjk0NjU3NmQtYzY5Yy00YWJkLWE5ZmYtZDNkYTkyMjU0NmYyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDExNDg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5Y2VmMGVhYWFkNGRlOWE0NzdiNDFlNzYyMjFmZWU3ZTBhYzQxNjIyMmMwODI0NGY4ZjU0YTc2Yjc0ZjNkYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Bv_in7fiimSuDIjKPBeHkgo3fLmX9YYs55LGfjuOipU)
> 
> At the moment besides manipulating the sequence card or imposing a different view, there's no way for us to reset the transform, so it would be good if we can add a button for that like we do for colour overrides etc.

I made an issue for the feature request https://github.com/3drepo/3drepo.io/issues/4636

**Review by carmenfan — CHANGES_REQUESTED** [review]
Hidden objects are detected when the objects are hidden by a sequence, but colour overrides are ignored (completely, can't even remove them!)
img-06-11-23-10-11

Model: https://issue-4608.dev.3drepo.io/v5/viewer/DevTeamWorkspace/72d935e5-d15f-4bc4-8433-fb6d79d57301/bc8f82ef-db39-4cb0-a4f9-9bb47aa14dc9

**Comment by Amantini1997** [comment]
> Hidden objects are detected when the objects are hidden by a sequence, but colour overrides are ignored (completely, can't even remove them!) 
> **[IMAGE]**
> Model: https://issue-4608.dev.3drepo.io/v5/viewer/DevTeamWorkspace/72d935e5-d15f-4bc4-8433-fb6d79d57301/bc8f82ef-db39-4cb0-a4f9-9bb47aa14dc9

I checked on staging and it seems the clearOverrides button is also not clearing geometry overrides from sequences.
I created a new issue to address this feature https://github.com/3drepo/3drepo.io/issues/4640

**Review by The-Daniel — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Pull request merged**